src/components/common/buttons/CustomButton.js

```diff
@@ -134,18 +162,22 @@ const CustomButton = (props: ButtonProps) => {
       contentStyle={styles.contentStyle}
       theme={{ ...theme, roundness: 50 }}
       uppercase={uppercase}
-      onPress={!props.loading && props.onPress}
+      disabled={disabled || loading}
+      onPress={props.onPress}
       {...buttonProps}
     >
-      {icon && (!iconAlignment || iconAlignment === 'left') && (
-        <IconButton icon={icon} theme={theme} dark={dark} size={iconSize} style={styles.leftIcon} />
-      )}
-      <TextContent dark={dark} uppercase={uppercase} textStyle={textStyle}>
-        {children}
-      </TextContent>
-      {icon && iconAlignment === 'right' && (
-        <IconButton icon={icon} theme={theme} dark={dark} size={iconSize} style={styles.rightIcon} />
-      )}
+      <View style={styles.contentWrapper}>
+        {icon && (!iconAlignment || iconAlignment === 'left') && (
+          <IconButton icon={icon} theme={theme} dark={dark} size={iconSize} style={styles.leftIcon} />
+        )}
+        {loading && <ActivityIndicator style={styles.activityIndicator} animating={loading} color={'#fff'} size={23} />}
```

Comment:
I think Activity Indicator should be adapted to the font size...because in some cases we could need to change size...same with color if the button has a light color

Reply:
@kbardi  Changed color to use light or props.color depending on button mode.
Didn't add an activity indicator size since there is only one size right now and it's not clear if this value should be calculated from other prop or we should add a new prop.